catalog/resource_sql_table.go

```diff
@@ -0,0 +1,259 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	clustersApi "github.com/databricks/databricks-sdk-go/service/clusters"
+	"github.com/databricks/terraform-provider-databricks/clusters"
+	"github.com/databricks/terraform-provider-databricks/common"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type SqlColumnInfo struct {
+	Name             string `json:"name"`
+	TypeText         string `json:"type_text"`
+	TypeJson         string `json:"type_json,omitempty"`
+	TypeName         string `json:"type_name"`
+	TypePrecision    int32  `json:"type_precision,omitempty"`
+	TypeScale        int32  `json:"type_scale,omitempty"`
+	TypeIntervalType string `json:"type_interval_type,omitempty"`
+	Position         int32  `json:"position"`
+	Comment          string `json:"comment,omitempty"`
+	Nullable         bool   `json:"nullable,omitempty" tf:"default:true"`
+	PartitionIndex   int32  `json:"partition_index,omitempty"`
+}
+
+type SqlTableInfo struct {
```

Comment:
SDK - https://github.com/databricks/databricks-sdk-go/blob/main/service/unitycatalog/model.go#L2124

Reply:
We decided internally to keep this model local until the unity API has full support for tables